### PR TITLE
Update github.com/spdx/gordf dependency to fix 32 bit support

### PR DESCRIPTION
**Pull request by kpcyrd**
This updates the gordf dependency so osv-scanner can be built for 32 bit targets. Related issues:

- https://github.com/spdx/gordf/pull/8
- https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/42405

## before

```
% GOARCH=386 go build ./cmd/osv-scanner/
# github.com/spdx/gordf/rdfloader/xmlreader
../../../go/pkg/mod/github.com/spdx/gordf@v0.0.0-20201111095634-7098f93598fb/rdfloader/xmlreader/xmlreader.go:213:27: WHITESPACE (untyped int constant 4294977024) overflows int
%
```

## after

```
% GOARCH=386 go build ./cmd/osv-scanner/
%
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -25,6 +25,8 @@ github.com/russross/blackfriday/v2 v2.1.0 h1:JIOH55/0cWyOuilr9/qlrm0BSXldqnqwMsf
 github.com/russross/blackfriday/v2 v2.1.0/go.mod h1:+Rmxgy9KzJVeS9/2gXHxylqXiyQDYRxCVz55jmeOWTM=
 github.com/spdx/gordf v0.0.0-20201111095634-7098f93598fb h1:bLo8hvc8XFm9J47r690TUKBzcjSWdJDxmjXJZ+/f92U=
```
Comment (inferno-chromium) [inline go.sum:26]: Are these 2 lines with older gordf reference still required ?

Comment (kpcyrd) [reply go.sum]: I don't know, running `go mod tidy` removed one of them but I don't know why the other one is still there. Manually deleting it doesn't work, running `go mod tidy` is adding it again. Updating all dependencies also doesn't get rid of it:
```
% go get -u ./...; go mod tidy; grep rdf go.sum 
github.com/spdx/gordf v0.0.0-20201111095634-7098f93598fb/go.mod h1:uKWaldnbMnjsSAXRurWqqrdyZen1R7kxl8TkmWk2OyM=
github.com/spdx/gordf v0.0.0-20221230105357-b735bd5aac89 h1:dArkMwZ7Mf2JiU8OfdmqIv8QaHT4oyifLIe1UhsF1SY=
github.com/spdx/gordf v0.0.0-20221230105357-b735bd5aac89/go.mod h1:uKWaldnbMnjsSAXRurWqqrdyZen1R7kxl8TkmWk2OyM=
```

**Review by oliverchang — APPROVED** [review]

**Comment by tuananh** [comment]
can you cut a release for this so that alpinelinux can package this?

https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/42405

**Comment by oliverchang** [comment]
> can you cut a release so that alpinelinux can package this?
> 
> https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/42405

We were planning to cut a release next week, or were you looking for something sooner?

**Comment by tuananh** [comment]
Thats fine. Thanks

**Pull request merged**